### ISSUE TITLE
 Spring boot Azure functions do not work properly when changing AzureSpringBootRequestHandler to FunctionsInvoker

### ISSUE THREAD

**Issue by m-moris** [spring-cloud/spring-cloud-function#692]
I have tried the following sample, and it works correctly as is.

* https://github.com/Azure-Samples/hello-spring-function-azure

**Correct result.**

```
$ curl http://localhost:7071/api/hello?name=Azure
{
  "message": "Hello, Azure!\n"
}
```

In this sample, AzureSpringBootRequestHandler is used, and since this class is deprecated, I rewrote it to FunctionInvoker and it does not work correctly.

```java

public class HelloHandler extends FunctionInvoker<User, Greeting> {

   
    @FunctionName("hello")
    public HttpResponseMessage execute(
            @HttpTrigger(name = "request", methods = {HttpMethod.GET, HttpMethod.POST}, authLevel = AuthorizationLevel.ANONYMOUS) HttpRequestMessage<Optional<User>> request,
            ExecutionContext context) {
        User user = request.getBody()
                .filter((u -> u.getName() != null))
                .orElseGet(() -> new User(
                        request.getQueryParameters()
                                .getOrDefault("name", "world")));
        context.getLogger().info("Greeting user name: " + user.getName());

        return request
                .createResponseBuilder(HttpStatus.OK)
                .body(handleRequest(user, context))
                .header("Content-Type", "application/json")
                .build();
    }
}
```

**Wrong result**
```
$ curl http://localhost:7071/api/hello?name=Azure
{
  "onErrorCall": {},
  "optimizableOperator": {},
  "source": {}
}
```

This sample also use Mono. I thinks handleRequests method will retrun a Mono object.

```java
@Component
public class HelloFunction implements Function<Mono<User>, Mono<Greeting>> {

    public Mono<Greeting> apply(Mono<User> mono) {
        return mono.map(user -> new Greeting("Hello, " + user.getName() + "!\n"));
    }
}
```

I output the log, and got `[2021-04-30T09:33:06.390Z] class reactor.core.publisher.MonoPeekTerminal`.

```java
        Object result =  handleRequest(user, context);
        context.getLogger().info(result.getClass().toString());
```

Thanks.

**Comment by konjp** [comment]
Any update on this? Is it possible to use Spring WebFlux with `FunctionInvoker`?

**Comment by olegz** [comment]
Indeed there was an issue but it is fixed now and will be available with the v3.1.3 which we're planning to release tomorrow. 
Also, the whole point of new FunctionInvoker is to eliminate all the code that you have above, so I also added the example - https://github.com/spring-cloud/spring-cloud-function/blob/main/spring-cloud-function-samples/function-sample-azure/src/main/java/example/ReactiveUppercaseHandler.java
